warnings building `llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp`

```
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:1706:6: warning: ‘bool llvm::operator==(const SDTypeConstraint&, const SDTypeConstraint&)’ has not been declared within ‘llvm’
 1706 | bool llvm::operator==(const SDTypeConstraint &LHS,
      |      ^~~~
In file included from /home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:14:
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h:390:15: note: only here as a ‘friend’
  390 |   friend bool operator==(const SDTypeConstraint &LHS,
      |               ^~~~~~~~
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:1732:6: warning: ‘bool llvm::operator<(const SDTypeConstraint&, const SDTypeConstraint&)’ has not been declared within ‘llvm’
 1732 | bool llvm::operator<(const SDTypeConstraint &LHS, const SDTypeConstraint &RHS) {
      |      ^~~~
/home/user/CLionProjects/llvm-project/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h:392:15: note: only here as a ‘friend’
  392 |   friend bool operator<(const SDTypeConstraint &LHS,
      |               ^~~~~~~~
```

I am using `gcc (GCC) 14.2.1 20240910` on Manjaro Linux.

--- a/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h
+++ b/llvm/utils/TableGen/Common/CodeGenDAGPatterns.h
@@ -393,6 +393,9 @@ struct SDTypeConstraint {
                         const SDTypeConstraint &RHS);
 };
 
+bool operator==(const SDTypeConstraint &LHS, const SDTypeConstraint &RHS);
+bool operator<(const SDTypeConstraint &LHS, const SDTypeConstraint &RHS);
+
 /// ScopedName - A name of a node associated with a "scope" that indicates
 /// the context (e.g. instance of Pattern or PatFrag) in which the name was
 /// used. This enables substitution of pattern fragments while keeping track

